### PR TITLE
Update mail-importer to work again

### PR DESCRIPTION
**Pull request by xylo04**
Update Java platform, dependencies and tests. Use `google_java_format`.

### CODE REVIEW

**Review by iflan — COMMENTED** [review]
Hi Chris,

This is excellent! Thank you for updating this to use the loopback address since the OOB method was deprecated.

I see that lots of the changes are just updating the order of imports and reformatting the documentation. I imagine that this is to conform with Google style these days. I also note that it expands some lines to 100 columns even though `CONTRIBUTING.md` says the current limit is 80.

Would you be willing to split this into two changes? The first would be all of the formatting changes (along with a change to `CONTRIBUTING.md`) and the second would be the substantive changes that actually make it work.

If that's too much work, let me know and I can try to do it. I think you've done the hardest part, which is just figuring out how to make it work again. :-)

Thanks again!

Ian

**Comment by xylo04** [comment]
> Would you be willing to split this into two changes? The first would be all of the formatting changes (along with a change to `CONTRIBUTING.md`) and the second would be the substantive changes that actually make it work.

Sure, I can do that. Would you prefer to see those as two separate PRs or just two commits in one PR?

File: Design.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ The Mail Importer tries to be efficient by not uploading the message content
 for messages that already exist in Gmail. This makes it easy for the import
 process to be restarted if there is an error.
 
-__Mail Importer__ requires Java 8 or later.
+**Mail Importer** requires Java 17 or later.
```
[the remark this answers is not in the document]
Comment (xylo04) [reply Design.md]: I think I had done this early on to see if any dependencies needed this, but it turns out they don't. Reverted this.

File: src/main/java/to/lean/tools/gmail/importer/gmail/Authorizer.java [hunk 2]
```diff
@@ -87,7 +85,7 @@ public Credential get() {
       // If we don't, prompt them to get one.
       if (credential == null) {
         String url = flow.newAuthorizationUrl()
-            .setRedirectUri(GoogleOAuthConstants.OOB_REDIRECT_URI)
+            .setRedirectUri("http://localhost:8080/oauth2callback")
```
[the remark this answers is not in the document]
Comment (xylo04) [reply src/main/java/to/lean/tools/gmail/importer/gmail/Authorizer.java]: Done.

**Comment by xylo04** [comment]
> Thank you for updating this to use the loopback address since the OOB method was deprecated.

Note that the way I'm doing this at the moment, I'm not actually starting a web server to capture that callback; I'm just letting the browser fail then copying the token out of the URL.

**Comment by iflan** [comment]
Thank you. This is excellent. If you want to improve it, adding a little bit about having the browser redirect fail and copying the token from the URL might be useful.

**Pull request merged**